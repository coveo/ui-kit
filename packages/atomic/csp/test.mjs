import {equal, notEqual} from 'node:assert';
import {fork} from 'node:child_process';
import {resolve} from 'node:path';
import {describe, it, before, after} from 'node:test';
import {launch} from 'puppeteer';

describe('CSP Smoketest', () => {
  let serverProcess;
  let browser;
  before(async () => {
    browser = await launch({headless: process.env.CI === 'true'});
    serverProcess = fork(resolve(import.meta.dirname, 'server.mjs'));
    await new Promise((resolve) => serverProcess.on('message', resolve));
  });

  after(async () => {
    serverProcess.kill();
    await browser.close();
  });

  it('should load the csp-ready page without errors', async () => {
    const page = await browser.newPage();
    const errors = [];
    page.on('console', (consoleMessage) => {
      if (consoleMessage.type() === 'error') {
        errors.push(consoleMessage);
      }
    });
    await page.goto('http://localhost:3000/examples/csp.html');
    await page.waitForNetworkIdle({waitForNetworkIdle: 1000});
    equal(errors.length, 0, 'No errors should be thrown on the page');
  });

  it('should load the non-csp-ready page with errors', async () => {
    const page = await browser.newPage();
    const errors = [];
    page.on('console', (consoleMessage) => {
      if (consoleMessage.type() === 'error') {
        errors.push(consoleMessage);
      }
    });
    await page.goto('http://localhost:3000/index.html');
    await page.waitForNetworkIdle({waitForNetworkIdle: 1000});
    notEqual(errors.length, 0, 'Errors should be thrown on the page');
  });
});
