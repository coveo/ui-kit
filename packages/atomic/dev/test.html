<!doctype html>
<html dir="ltr" lang="en">

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
    <title>E2E tests</title>

    <script type="module">
      import { defineCustomElements } from '@coveo/atomic/loader';
      import '@coveo/atomic/themes/accessible.css';

      defineCustomElements();
    </script>

    <style>
      ::part(ripple) {
        opacity: 10%;
      }
    </style>
  </head>

  <body></body>

</html>
