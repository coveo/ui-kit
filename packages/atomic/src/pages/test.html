<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
    <title>E2E tests</title>

    <script type="module" src="/build/atomic.esm.js"></script>
    <script nomodule src="/build/atomic.js"></script>
  </head>
  <body>
    <h1>Content for Integration Test will be replaced here.</h1>
  </body>
</html>
