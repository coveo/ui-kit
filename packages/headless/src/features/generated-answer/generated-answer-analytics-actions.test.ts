import {
  buildMockSearchAppEngine,
  MockSearchEngine,
} from '../../test/mock-engine';
import {buildMockSearchResponse} from '../../test/mock-search-response';
import {buildMockSearchState} from '../../test/mock-search-state';
import {createMockState} from '../../test/mock-state';
import {
  logGeneratedAnswerDetailedFeedback,
  logGeneratedAnswerFeedback,
} from './generated-answer-analytics-actions';

const mockMakeGeneratedAnswerFeedbackSubmit = jest.fn();

jest.mock('coveo.analytics', () => {
  const mockCoveoSearchPageClient = jest.fn(() => ({
    disable: jest.fn(),
    makeGeneratedAnswerFeedbackSubmit: mockMakeGeneratedAnswerFeedbackSubmit,
  }));

  return {
    CoveoSearchPageClient: mockCoveoSearchPageClient,
    history: {HistoryStore: jest.fn()},
  };
});

const exampleFeedback = 'irrelevant';
const exampleGenerativeQuestionAnsweringId = '123';
const exampleDetails = 'example details';

describe('the analytics related to the generated answer feature', () => {
  let engine: MockSearchEngine;

  beforeEach(() => {
    engine = buildMockSearchAppEngine({
      state: createMockState({
        search: buildMockSearchState({
          response: buildMockSearchResponse({
            extendedResults: {
              generativeQuestionAnsweringId:
                exampleGenerativeQuestionAnsweringId,
            },
          }),
        }),
      }),
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should log #logGeneratedAnswerFeedback with the right payload', async () => {
    await engine.dispatch(logGeneratedAnswerFeedback(exampleFeedback));

    const mockToUse = mockMakeGeneratedAnswerFeedbackSubmit;
    const expectedMetadata = {
      generativeQuestionAnsweringId: exampleGenerativeQuestionAnsweringId,
      reason: exampleFeedback,
    };

    expect(mockToUse).toBeCalledTimes(1);
    expect(mockToUse).toBeCalledWith(expectedMetadata);
  });

  it('should log #logGeneratedAnswerDetailedFeedback with the right payload', async () => {
    await engine.dispatch(logGeneratedAnswerDetailedFeedback(exampleDetails));

    const mockToUse = mockMakeGeneratedAnswerFeedbackSubmit;
    const expectedMetadata = {
      generativeQuestionAnsweringId: exampleGenerativeQuestionAnsweringId,
      reason: 'other',
      details: exampleDetails,
    };

    expect(mockToUse).toBeCalledTimes(1);
    expect(mockToUse).toBeCalledWith(expectedMetadata);
  });
});
