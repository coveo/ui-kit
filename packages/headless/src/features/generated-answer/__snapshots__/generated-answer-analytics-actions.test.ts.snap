// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logCopyGeneratedAnswer with the right payload 1`] = `
[
  "Qna.AnswerAction",
  {
    "action": "copyToClipboard",
    "answer": {
      "id": "123",
      "type": "RGA",
    },
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logDislikeGeneratedAnswer with the right payload 1`] = `
[
  "Qna.SubmitFeedback",
  {
    "answer": {
      "id": "123",
      "type": "RGA",
    },
    "feedback": {
      "liked": false,
    },
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logGeneratedAnswerDetailedFeedback with the right payload 1`] = `
[
  "Qna.SubmitFeedback",
  {
    "answer": {
      "id": "123",
      "type": "RGA",
    },
    "feedback": {
      "details": "example details",
      "liked": false,
      "reason": "other",
    },
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logGeneratedAnswerHideAnswers with the right payload 1`] = `
[
  "Qna.AnswerAction",
  {
    "action": "hide",
    "answer": {
      "id": "123",
      "type": "RGA",
    },
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logGeneratedAnswerShowAnswers with the right payload 1`] = `
[
  "Qna.AnswerAction",
  {
    "action": "show",
    "answer": {
      "id": "123",
      "type": "RGA",
    },
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logHoverCitation with the right payload 1`] = `
[
  "Qna.CitationHover",
  {
    "answer": {
      "id": "123",
      "type": "RGA",
    },
    "citation": {
      "id": "some-citation-id",
      "type": "Source",
    },
    "citationHoverTimeInMs": 1234,
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logLikeGeneratedAnswer with the right payload 1`] = `
[
  "Qna.SubmitFeedback",
  {
    "answer": {
      "id": "123",
      "type": "RGA",
    },
    "feedback": {
      "liked": true,
    },
  },
]
`;

exports[`generated answer analytics actions when analyticsMode is \`next\` should log #logOpenGeneratedAnswerSource with the right payload 1`] = `
[
  "Qna.CitationClick",
  {
    "answer": {
      "id": "123",
      "type": "RGA",
    },
    "citation": {
      "id": "some-citation-id",
      "type": "Source",
    },
  },
]
`;
